instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/patches/active_job_callbacks.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActiveJob
+      module Patches
+        # Module to prepend to ActiveJob::Base for instrumentation.
+        module ActiveJobCallbacks
+          def self.prepended(base) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity
+            base.class_eval do # rubocop:disable Metrics/BlockLength
+              around_enqueue do |job, block|
+                span_kind = job.class.queue_adapter_name == 'inline' ? :client : :producer
+                span_name = "#{otel_config[:job_class_span_names] ? job.class : job.queue_name} send"
+                span_attributes = job_attributes(job)
+                otel_tracer.in_span(span_name, attributes: span_attributes, kind: span_kind) do
+                  OpenTelemetry.propagation.inject(job.metadata)
+                  block.call
+                end
+              end
+
+              around_perform do |job, block|
+                span_kind = job.class.queue_adapter_name == 'inline' ? :server : :consumer
+                span_name = "#{otel_config[:job_class_span_names] ? job.class : job.queue_name} process"
+                span_attributes = job_attributes(job).merge('messaging.operation' => 'process')
+
+                propagated_context = OpenTelemetry.propagation.extract(job.metadata)
+
+                if otel_config[:propagation_style] == :link
+                  span_links = [
+                    OpenTelemetry::Trace::Link.new(OpenTelemetry::Trace.current_span(propagated_context).context)
+                  ]
+                end
+
+                parent_context = (otel_config[:propagation_style] == :child ? propagated_context : OpenTelemetry::Context.current)
+                OpenTelemetry::Context.with_current(parent_context) do
+                  otel_tracer.in_span(span_name, attributes: span_attributes, links: span_links, kind: span_kind) do |span|
+                    span.set_attribute('messaging.active_job.executions', job.executions)
+                    block.call
+                  end
+                end
```

Comment:
I think the instrumentation patch here will probably need the same treatment as the resque PR did https://github.com/open-telemetry/opentelemetry-ruby/pull/779#discussion_r645895665

The TLDR is we lose baggage propagation when using `:link` or `:none`

Reply:
This should be working now.